File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/lightdata/ui/LightDataPresenterModule.java
package com.fintechplatform.ui.profile.lightdata.ui;

import com.fintechplatform.ui.models.DataAccount;
import com.fintechplatform.api.user.api.ProfileAPI;
import com.fintechplatform.ui.profile.db.user.UsersPersistanceDB;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class LightDataPresenterModule {

    private LightDataContract.View view;
    private DataAccount configuration;

    public LightDataPresenterModule(LightDataContract.View view, DataAccount configuration) {
        this.view = view;
        this.configuration = configuration;
    }

    @Provides
    @Singleton
    LightDataContract.Presenter providesLightDataPresenter(ProfileAPI api, UsersPersistanceDB usersPersistanceDB) {
        return new LightDataPresenter(view, api, configuration, usersPersistanceDB);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/lightdata/di/LightDataPresenterModule.kt
package com.fintechplatform.ui.profile.lightdata.di

import com.fintechplatform.api.profile.api.ProfileAPI
import com.fintechplatform.ui.models.DataAccount
import com.fintechplatform.ui.profile.db.user.UsersPersistanceDB
import com.fintechplatform.ui.profile.lightdata.LightDataContract
import com.fintechplatform.ui.profile.lightdata.LightDataPresenter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class LightDataPresenterModule(private val view: LightDataContract.View, private val configuration: DataAccount) {

    @Provides
    @Singleton
    internal fun providesLightDataPresenter(api: ProfileAPI, usersPersistanceDB: UsersPersistanceDB): LightDataContract.Presenter {
        return LightDataPresenter(view, api, configuration, usersPersistanceDB)
    }
}
